packages/wallet/test/SingleAddressWallet/load.test.ts

```diff
@@ -390,3 +391,31 @@ describe('SingleAddressWallet creates big UTXO', () => {
     wallet.shutdown();
   });
 });
+
+describe('SingleAddressWallet.error$', () => {
+  it('emits non retryable errors', async () => {
+    const stores = createInMemoryWalletStores();
+    const utxoSet = generateUtxos(30, 10);
+
+    const wallet = await createWallet(stores, {
+      chainHistoryProvider: mocks.mockChainHistoryProvider(),
+      networkInfoProvider: {
+        ...mocks.mockNetworkInfoProvider(),
+        ledgerTip: jest.fn().mockRejectedValue(new InvalidStringError('Test invalid string error'))
+      },
+      rewardsProvider: mocks.mockRewardsProvider(),
+      utxoProvider: mocks.mockUtxoProvider(utxoSet)
+    });
+
+    try {
+      await firstValueFrom(wallet.tip$);
+      expect(false).toBeTruthy();
```

Comment:
I think either `expect.assertions(n)` or this
```suggestion
      throw new Error('Expected tip$' to throw)
```
would be a little clearer.

Also, do we want to throw on both `error$` and other observables? When I suggested `error$`, I thought it would be a 'catch-all' and other observables wouldn't throw. If we throw on other observables then maybe we don't need `error$`? Thoughts @rhyslbw @mirceahasegan ?

Reply:
ffbc1017a should solve this